### DISCUSSION TITLE
Proposal: Support new Python 3.7 data class mechanism

### DISCUSSION THREAD

**Discussion by mfhepp** [RDFLib/rdflib#1569] [Ideas]
Dear:
I think the new data class mechanism in Python 3.7 would be a great fit for manipulating RDF data in Python:

https://realpython.com/python-data-classes/

Martin

**Comment by ashleysommer** [comment]
Can you elaborate on your proposal? Specifically, how would python 3.7 dataclasses benefit the RDFLib use-case of working with RDF data in python?

Also, being a 3.7+ only feature, implementing this feature would remove support for python 3.4, 3.5, and 3.6, which are the most commonly used python versions.
